From: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>
Date: Wed, 29 May 2024 22:53:30 -0700
Subject: cpufreq: intel_pstate: Support Emerald Rapids OOB mode
Git-commit: 7e1c3f584ee78b0d0210fc424420d9529f3ca952
Patch-mainline: v6.11-rc1
References: jsc#PED-10835

Prevent intel_pstate from loading when OOB (Out Of Band) P-states mode is
enabled in Emerald Rapids.

The OOB identifying bits are same as for the prior generation CPUs
like Sapphire Rapids servers, so also add Emerald Rapids to the
intel_pstate_cpu_oob_ids[] list.

Signed-off-by: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Acked-by: Giovanni Gherdovich <ggherdovich@suse.cz>
---
 drivers/cpufreq/intel_pstate.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/cpufreq/intel_pstate.c
+++ b/drivers/cpufreq/intel_pstate.c
@@ -2419,6 +2419,7 @@ static const struct x86_cpu_id intel_pst
 	X86_MATCH(INTEL_SKYLAKE_X,		core_funcs),
 	X86_MATCH(INTEL_ICELAKE_X,		core_funcs),
 	X86_MATCH(INTEL_SAPPHIRERAPIDS_X,	core_funcs),
+	X86_MATCH(INTEL_EMERALDRAPIDS_X,	core_funcs),
 	{}
 };
 
